src/provable/curve-bigint.ts

```diff
@@ -108,6 +111,10 @@ const PublicKey = {
   toInputLegacy({ x, isOdd }: PublicKey): HashInputLegacy {
     return { fields: [x], bits: [!!isOdd] };
   },
+
+  toFields({ x, isOdd }: PublicKey) {
+    return [x, isOdd ? 1n : 0n];
```

Comment:
`PublicKey.toFields` was returning `[y, x]` instead of `[x, y]` like SnarkyJS does. It's probably better to reverse the order on the SnarkyJS side, to not introduce any breaking changes into mina-signer or do you think thats not the case @mitschabaude ?

Reply:
ph -- I think this is a bug, it should be `[x, y]`

this is the change you probably need (a couple of lines above this):
```diff
- ...provable({ x: Field, isOdd: Bool }),
+ ...provable({ x: Field, isOdd: Bool }, { customObjectKeys: ['x', 'isOdd'] }),
```

btw, since I wrote the binable code I learned that the order in JS objects is deterministic and basically the insertion order, so I might be able to get rid of `customObjectKeys`